homeassistant/components/nest/config_flow.py

```diff
@@ -66,6 +67,23 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+class ConfigMode(Enum):
+    """Integration configuration mode."""
+
+    SDM = 1  # SDM api with configuration.yaml
+    LEGACY = 2  # "Works with Nest" API
+
+
+def get_config_mode(hass: HomeAssistant) -> ConfigMode:
+    """Return the integration configuration mode."""
+    if DOMAIN not in hass.data:
+        return ConfigMode.SDM
```

Comment:
Note to reviewer: In the future, this case will return ConfigMode.SDM_CONFIG_ENTRY to use a pure config flow without configuration.yaml